jupyterlab_polus_render/README.md

```diff
@@ -1,119 +1,152 @@
+```bash
+conda create -n jupyterlab_polus_render-dev -c conda-forge nodejs python jupyterlab=4.0.11
+conda activate jupyterlab_polus_render-dev
+```
+
+```bash
+# Clone the repo to your local environment
+# Change directory to the jupyterlab_polus_render directory
+# Install package in development mode
+pip install -e .
+# Link your development version of the extension with JupyterLab
+jupyter labextension develop . --overwrite
+# Server extension must be manually installed in develop mode
+jupyter server extension enable jupyterlab_polus_render
+# Rebuild extension Typescript source after making changes
+jlpm run build
 ```
 
-# Functions
-``` Python
-def render(nbhub_url:ParseResult, nb_root:PurePath = Path(os.getenv('HOME')) if "HOME" in os.environ else Path("/home/jovyan/"),    
-           image_location:Union[ParseResult, PurePath] = "", 
-           microjson_overlay_location:Union[ParseResult, PurePath] = "", width:int=960, height:int=500, use_static:bool = True)->str:
-    """
-    Embeds Polus Render into a JupyterLabs notebook with the help of `render-server-ext`
-
-    Param:
-        nbhub_url (ParseResult): URL used used for jupyterhub. Contains '/lab/' in its uri
-        nb_root (PurePath): Root path used to search files in. Default is os.getenv('HOME') else \"/home/joyvan/\""
-                            if HOME does not exist.
-                if absolute paths will be used for images and json files.
-        image_location(ParseResult|Purepath): Acquired from urllib.parse.ParseResult or Path, renders url in render.
-                            If not specified, renders default render url.
-        microjson_overlay_location(ParseResult|Purepath): Acquired from urllib.parse.ParseResult or Path, renders url in render.
-                            If not specified, renders default render url
-        width (int): width of render to be displayed, default is 960
-        height (int): height of render to be displayed, default is 500
-        use_static (bool): Use static build of render, default is True
-    Returns: Render URL
-    """
+### How to see your changes
+#### Typescript:
+If you use JupyterLab to develop then you can watch the source directory and run JupyterLab at the same time in different
+terminals to watch for changes in the extension's source and automatically rebuild the widget.
+
+```bash
+# Watch the source directory in one terminal, automatically rebuilding when needed
+jlpm run watch
+# Run JupyterLab in another terminal
+jupyter lab
+```
+
+After a change wait for the build to finish and then refresh your browser and the changes should take effect.
+
+#### Python:
+If you make a change to the python code then you will need to restart the notebook kernel to have it take effect.
+
+
+### Development uninstall
+
+```bash
+# Server extension must be manually disabled in develop mode
+jupyter server extension disable jupyterlab_polus_render
+pip uninstall jupyterlab_polus_render
 ```
 
-# Implementation Details
```

Comment:
Could you add back the
# Implementation Details
section?

You would want to talk about what we use in the frontend (npm package) and that we serve images using the server extension